packages/renderers/renderer-solid/server.js

```diff
@@ -1,24 +1,23 @@
-import { createComponent } from 'solid-js';
-import { renderToStringAsync, ssr } from 'solid-js/web/dist/server.js';
+import { renderToString, ssr, createComponent } from 'solid-js/web/dist/server.js';
 
-async function check(Component, props, children) {
+function check(Component, props, children) {
   if (typeof Component !== 'function') return false;
 
-  const { html } = await renderToStaticMarkup(Component, props, children);
+  const { html } = renderToStaticMarkup(Component, props, children);
   return typeof html === 'string';
 }
 
-async function renderToStaticMarkup(Component, props, children) {
-  const html = await renderToStringAsync(
-    () => () =>
+function renderToStaticMarkup(Component, props, children) {
+  const html = renderToString(
+    () =>
       createComponent(Component, {
         ...props,
         // In Solid SSR mode, `ssr` creates the expected structure for `children`.
         // In Solid client mode, `ssr` is just a stub.
-        children: ssr([`<astro-fragment>${children}</astro-fragment>`]),
+        children: ssr(`<astro-fragment>${children}</astro-fragment>`),
       })
   );
-  return { html };
+  return { html: `<script>window._$HYDRATION||(window._$HYDRATION={events:[],completed:new WeakSet})</script>` + html };
```

Comment:
Can this be reversed so that the script is last? The reason is that `client:visible` works by observing children of the root and script tags don't trigger IntersectionObserver callbacks. Putting the script at the end fixes it. 

It won't affect hydration because Astro islands are lazy anyways, this script runs before hydration even if it's at the end.

Reply:
Ah.. that's what I hit in the stream. Ok sure no problem I will change it.